source/extensions/transport_sockets/tls/utility.cc

```diff
@@ -253,16 +253,16 @@ std::string Utility::getSubjectFromCertificate(X509& cert) {
   return getRFC2253NameFromCertificate(cert, CertName::Subject);
 }
 
-int32_t Utility::getDaysUntilExpiration(const X509* cert, TimeSource& time_source) {
+absl::optional<size_t> Utility::getDaysUntilExpiration(const X509* cert, TimeSource& time_source) {
   if (cert == nullptr) {
     return std::numeric_limits<int>::max();
   }
   int days, seconds;
   if (ASN1_TIME_diff(&days, &seconds, currentASN1_Time(time_source).get(),
                      X509_get0_notAfter(cert))) {
-    return days;
+    return days > 0 ? absl::optional<size_t>(days) : absl::nullopt;
```

Comment:
Should it be greater-equals?, i.e.:
```suggestion
    return days >= 0 ? absl::optional<size_t>(days) : absl::nullopt;
```

Reply:
Sure.